src/Export/VTK.jl

```diff
@@ -111,46 +171,167 @@ function component_names(::Type{S}) where S
+"""
+function write_celldata(vtks::VTKStream, celldata, name)
+    WriteVTK.vtk_cell_data(vtks.vtk, celldata, name)
+end
+
+"""
+    write_nodedata(vtks::VTKStream, nodedata::Vector{Real}, name)
+    write_nodedata(vtks::VTKStream, nodedata::Vector{<:AbstractTensor}, name)
+    
+Write the `nodedata` that is ordered by the nodes in the grid to the vtk file.
+
+When `nodedata` contains `Tensors.Vec`s, each component is exported. 
+Two-dimensional vectors are padded with zeros.
+
+When `nodedata` contains second order tensors, the index order, 
+`[11, 22, 33, 23, 13, 12, 32, 31, 21]`, follows the default Voigt order in Tensors.jl.
+"""
+function write_nodedata(vtks::VTKStream, nodedata, name)
+    _vtk_write_nodedata(vtks.vtk, nodedata, name)
+end
+
+
+"""
+    write_nodeset(vtks::VTKStream, nodeset::String)
+
+Export nodal values of 1 for nodes in `nodeset`, and 0 otherwise
+"""
+function write_nodeset(vtks::VTKStream, nodeset::String)
+    grid = getgrid(vtks)
     z = zeros(getnnodes(grid))
     z[collect(getnodeset(grid, nodeset))] .= 1.0
-    vtk_point_data(vtk, z, nodeset)
+    WriteVTK.vtk_point_data(vtks.vtk, z, nodeset)
+    return vtks
 end
 
 """
-    vtk_cellset(vtk, grid::Grid)
+    write_cellset(vtks)
+    write_cellset(vtks, cellset::String)
+    write_cellset(vtks, cellsets::Union{AbstractVector{String},AbstractSet{String})
 
-Export all cell sets in the grid. Each cell set is exported with
-`vtk_cell_data` with value 1 if the cell is in the set, and 0 otherwise.
+Export all cell sets in the grid with name according to their keys and 
+celldata 1 if the cell is in the set, and 0 otherwise. It is also possible to 
+only export a single `cellset`, or multiple `cellsets`. 
 """
-function vtk_cellset(vtk::WriteVTK.DatasetFile, grid::AbstractGrid, cellsets=keys(grid.cellsets))
+function write_cellset(vtks, cellsets=keys(getcellsets(getgrid(vtks))))
+    grid = getgrid(vtks)
     z = zeros(getncells(grid))
     for cellset in cellsets
         z .= 0.0
         z[collect(getcellset(grid, cellset))] .= 1.0
-        vtk_cell_data(vtk, z, cellset)
+        write_celldata(vtks, z, cellset)
     end
-    return vtk
+    return vtks
 end
+write_cellset(vtks::VTKStream, cellset::String) = write_cellset(vtks, [cellset])
 
 """
-    vtk_cellset(vtk, grid::Grid, cellset::String)
+    write_dirichlet(vtks::VTKStream, ch::ConstraintHandler)
 
-Export the cell set specified by `cellset` as cell data with value 1 if
-the cell is in the set and 0 otherwise.
+Saves the dirichlet boundary conditions to a vtkfile.
+Values will have a 1 where bcs are active and 0 otherwise
 """
-vtk_cellset(vtk::WriteVTK.DatasetFile, grid::AbstractGrid, cellset::String) =
-    vtk_cellset(vtk, grid, [cellset])
+function write_dirichlet(vtks, ch::ConstraintHandler)
+    
+    if getgrid(vtks) !== ch.dh.grid 
+        @warn("The grid saved in $(typeof(vtks)) and ConstraintHandler are not aliased, no checks are performed to ensure that they are equal")
```

Comment:
Why not error?

Reply:
In case a user (for some reason I cannot think of) needs to use unaliased meshes. But let's error for now, and change to warning if a need arises.